dev-util/hip/hip-5.3.3-r4.ebuild

```diff
@@ -97,11 +98,16 @@ src_prepare() {
 	hprefixify $(grep -rl --exclude-dir=build/ --exclude="hip-config.cmake.in" "/usr" "${S}")
 	hprefixify $(grep -rl --exclude-dir=build/ --exclude="hipcc.pl" "/usr" "${HIP_S}")
 
-	cp "$(prefixify_ro "${FILESDIR}"/hipvars-5.3.3.pm)" bin/hipvars.pm || die "failed to replace hipvars.pm"
-	sed -e "s,@HIP_BASE_VERSION_MAJOR@,$(ver_cut 1)," -e "s,@HIP_BASE_VERSION_MINOR@,$(ver_cut 2)," \
-		-e "s,@HIP_VERSION_PATCH@,$(ver_cut 3)," \
-		-e "s,@CLANG_PATH@,${LLVM_PREFIX}/bin," -i bin/hipvars.pm || die
-	popd || die
+	mv bin/hipvars.pm "${T}"/hipvars.pm.orig
+	cp "$(prefixify_ro "${FILESDIR}"/hipvars.pm)" bin/hipvars.pm || die "failed to replace hipvars.pm"
+	sed -n '/^\$HIP_BASE_VERSION_.* = "/p' "${T}"/hipvars.pm.orig >> bin/hipvars.pm || die
+	cat <<-EOF >> bin/hipvars.pm || die
+	\$HIP_VERSION_MAJOR="\$HIP_BASE_VERSION_MAJOR";
+	\$HIP_VERSION_MINOR="\$HIP_BASE_VERSION_MINOR";
+	\$HIP_VERSION_PATCH="\$HIP_BASE_VERSION_PATCH";
+	\$HIP_VERSION="\$HIP_BASE_VERSION_MAJOR.\$HIP_BASE_VERSION_MINOR.\$HIP_BASE_VERSION_PATCH";
+	EOF
+	sed	-e "s,@CLANG_PATH@,${LLVM_PREFIX}/bin," -i bin/hipvars.pm || die
```

Comment:
This logic is too complex to figure out what you are trying to fix hipvars.pm.  Is it possible to simplify it?

Reply:
Maybe I can put the lines appended in another file. But overall that does not simplify much.

Actually, I'd like to try the upstream `hipvars.pm`. Maybe after several versions of refining it become easy to adopt.